components/workshops.tsx

```diff
@@ -1,9 +1,80 @@
-const Workshops = () => (
-  <div className="bg-red-400">
-    <div className="flex flex-col sm:flex-row p-6 sm:p-12">
-      <h1 className="text-5xl sm:text-9xl font-bold">2. Workshops</h1>
+import WorkshopCard from "./card"
+import { useState, useEffect } from "react";
+import IEvent from "../utils/IEvent";
+
+const Workshops = () => {
+  const [events, setEvents] = useState<IEvent[]>([]);
+
+  const fetchEvents = async() => {
+    const response = await fetch('/api/workshop', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+    });
+    console.log(response)
+
+    if (!response.ok) {
+      throw new Error(`Error: ${response.status}`);
```

Comment:
This code will almost never be run so it's not a big deal, but I think it's bad practice to throw an error to the client. If the API returns an error, then the page should instead load some fallback data or render a component that shows that something went wrong. IMO, throwing a Next.js error to the user is bad UX.